### DISCUSSION TITLE
Why does this happen? The same code. Left side can report errors on the right side

### DISCUSSION THREAD

**Discussion by wosiyourfather** [AvaloniaUI/Avalonia#12682] [Q&A]
![why](https://github.com/AvaloniaUI/Avalonia/assets/53590491/4e9618de-cfce-4f5d-bc39-413ad1d819e1)

**Comment by stevemonaco** [comment]
Does the project on the right use `<AvaloniaUseCompiledBindingsByDefault>true</AvaloniaUseCompiledBindingsByDefault>` in the csproj and the one on the left doesn't? If so, then it's because `Binding` on the left would default to `ReflectionBinding` instead of `CompiledBinding`.

Either way, you should assign an `x:DataType` on the right project if you're trying to bind through a `DataContext`. Probably at the `<ControlTheme>` level. Otherwise, use `ReflectionBinding` if this is some arbitrary type.

**Comment by timunie** [comment]
please read this: https://docs.avaloniaui.net/docs/next/basics/data/data-binding/compiled-bindings
